### DISCUSSION TITLE
SslStream.AuthenticateAsClient correct way to avoid sending SNIs

### DISCUSSION THREAD

**Discussion by Evengard** [dotnet/runtime#97443] [Q&A]
Hello!

In some cases, it is needed to construct an SslStream which doesn't send any SNIs to the target host. It seems this is possible by setting the targetHost parameter to an empty string, but we wonder if there is any side effects associated with that? Won't that break any other certificate checks under the hood? How safe is it to just omit targetHost for disabling SNI? Or maybe there are any other ways to do it?

Also, while in the documentation it is stated that TargetHost can't be null, the existing code seems to accept null without a problem. Is it intended behaviour with documentation being slightly out of date, or just a mishap?

**Comment by roji** [comment]
Note that re Host nullability, this isn't only a documentation quesiton, is an NRT annotation question.

**Comment by Evengard** [comment]
Migrated to issue https://github.com/dotnet/runtime/issues/97444
